src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaRepository.java

```diff
@@ -99,6 +94,30 @@ public void createTopic(final String topic, final int partitionsNum, final int r
         }
     }
 
+    public boolean topicExists(final String topic) throws NakadiException {
+        return listTopics()
+                .stream()
+                .map(Topic::getName)
+                .anyMatch(t -> t.equals(topic));
+    }
+
+    public boolean areCursorsCorrect(final String topic, final List<Cursor> cursors) {
```

Comment:
probably something in this direction would be indeed better, specially looking at the call site: 
```
// check that offsets are not out of bounds
if (cursors.isPresent() && !topicRepository.areCursorsCorrect(topic, cursors.get())) {
```

something like your suggestion would be immediately clear.